backend/src/main/java/de/tobias/intestinalinspector/controller/FoodController.java

```diff
@@ -48,15 +48,24 @@ public FoodController(FoodService foodService) {
     }
 
     @PutMapping
-    public ResponseEntity<Integer> update(@RequestBody long id, String newName){
-        int numberOfChangedRows = foodService.update(id, newName);
-        if(numberOfChangedRows == 1){
-            return ok(numberOfChangedRows);
+    public ResponseEntity<FoodDto> update(@RequestBody UpdateDto updateDto){
+        FoodEntity foodEntity = map(updateDto);
+        FoodEntity changedEntity = foodService.update(foodEntity);
+        if(changedEntity != null){
```

Comment:
wenn der update nicht klappt würde ich eine exception in der update methode werfen anstatt null zurück zu geben. die exception kannst du im exceptionhandler fangen und dort eine response mit status code und message zurück geben.